connector/extensions/dcp-impl/src/main/java/org/oaebudt/edc/dcp/policy/TrustedGroupCredentialEvaluationFunction.java

```diff
@@ -28,11 +28,11 @@ public boolean evaluate(Operator operator, Object rightValue, Permission rule, C
 
         var participantAgent = policyContext.participantAgent();
 
-        Set<String> trustedParticipants = participantGroupStore.findById(rightValue.toString()).participants();
-        if (Objects.isNull(trustedParticipants)) {
+        ParticipantGroup participantGroup = participantGroupStore.findById(rightValue.toString());
+        if (Objects.isNull(participantGroup)) {
             return false;
         }
 
-        return trustedParticipants.contains(participantAgent.getIdentity());
+        return participantGroup.participants().contains(participantAgent.getIdentity());
```

Comment:
```suggestion
        return participantGroup.contains(participantAgent.getIdentity());
```
the participant group could offer the `contains` method directly ([tell don't ask principle](https://martinfowler.com/bliki/TellDontAsk.html))

Reply:
Resolved